### PR TITLE
Update samples for readonly keyword

### PR DESCRIPTION
**Pull request by BillWagner**
Move existing snippets from docs repo.
Add samples for new uses of readonly.

For dotnet/docs#5682

### CODE REVIEW

File: snippets/csharp/keywords/ReadonlyKeywordExamples.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (rpetrusha) [inline snippets/csharp/keywords/ReadonlyKeywordExamples.cs]: nit: this and the System.Text namespace aren't actually used in the example, so I'd remove them.

**Comment by mairaw** [comment]
should we also remove the old ones that readonly would rely on once I got to that topic to make the switch to dotnet/samples? let me find where that would be.

**Comment by mairaw** [comment]
They are here: https://github.com/dotnet/samples/blob/master/snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsModifiers/CS/csrefKeywordsModifiers.cs 

I made as far as letter C on the keywords folder to remove the inline snippets 😄

**Comment by BillWagner** [comment]
@mairaw In dotnet/docs#6086 I removed the snippets from docs that were in this topic.

I've been trying to remove any as I move them. We can start a separate task to remove other unused snippets in that repo, but that feels like a lower priority.

**Comment by mairaw** [comment]
No, no... what I meant is that the sample already existed here and now we have two of the same (or almost the same).

**Pull request merged**